src/ast/transform/SemanticChecker.cpp

```diff
@@ -504,51 +504,41 @@ void SemanticCheckerImpl::checkClause(const Clause& clause) {
 }
 
 void SemanticCheckerImpl::checkComplexRule(const std::set<const Clause*>& multiRule) {
```

Comment:
Basically `std::set` does not guarantee that `Clause`s will be ordered in a deterministic way because it is ordered by pointer's value.
By chance it seems that on Linux and MacOS the clauses are allocated memory location with the same order as the lexical order of clauses in the rule.
Unfortunately on Visual Studio, the clauses memory locations are not deterministically ordered.